src/Bedrock.Framework.Experimental/Transports/Pooling/IConnectionPool.cs

```diff
@@ -0,0 +1,13 @@
+using System.Collections.Concurrent;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+
+namespace Bedrock.Framework
+{
+    public interface IConnectionPool
```

Comment:
We don't need this interface. After looking at this I think the IConnectionPool should be an IConnectionFactory and there should be a way to plug in a policy for the pool. Similar to the object pool that exists in https://github.com/dotnet/extensions/tree/master/src/ObjectPool/src. For now, we can remove this interface.

Reply:
Yeah, I mostly was following your suggestion of having:
```c#
builder.UsePooling(pool);
```

I'm on the same page as you; this should just be a factory. 

For providing a policy, it seems like the runtime has an abstract ObjectPool class, should we do the same for the policy provided for pooling?

Reply:
Lets design this in an issue outside of this PR, it'll require a few rounds.